src/main/java/seedu/address/model/person/Schedule.java

```diff
@@ -99,7 +99,17 @@ public void add(EventTime eventTime) throws SchedulingException {
         EventTime now = new EventTime(timeNow, timeNow);
         schedule.add(now);
 
-        NavigableSet<EventTime> candidates = schedule.subSet(now, true, lastCandidate, true);
+        // this should always be non null, since an event starting at midnight will always be the smallest
+        EventTime firstCandidate = schedule.lower(now);
+
+        if (!firstCandidate.overlaps(now)) {
+            firstCandidate = now;
+        } else {
+            schedule.remove(now);
+        }
```

Comment:
to compare in treeset, theres no need to add the now into the schedule. when you use schedule.lower(now), it will return you the previous eventTime even if u didn't add the now into the schedule